test/lib/DeclarativeTransforms/TestLinalgTransformPatterns.td

```diff
@@ -126,4 +125,26 @@ def : Pat<(IndexedGenericOp:$op $_1, $_2, $_3, $_4, $_5, $_6, $_7, $_8),
               [(Constraint<And<[HasNoLinalgTransformMarker,
                            AffineMapDomainHasDim<3>]>> $op)]>;
 
+//===----------------------------------------------------------------------===//
+// Linalg subview operands promotion.
+//===----------------------------------------------------------------------===//
+def : Pat<(MatmulOp:$op $A, $B, $C),
+          (LinalgOpPromoteSubviews<"MatmulOp"> $op),
+          [(Constraint<HasOperandsOfType<"SubViewOp">> $op),
+          (Constraint<HasLinalgTransformMarker<"_promote_views_">> $op)]>;
+
+def : Pat<(MatvecOp:$op $A, $b, $c),
```

Comment:
These patterns are not used, could we drop them?

Reply:
Is it worth keeping them and adding some tests for them?

Reply:
I wouldn't think so, it's very easy to copy-paste the MatmulOp and adapt.

Reply:
OK. Done.